### PR TITLE
Send emails using communication

### PR DESCRIPTION
**Pull request by monassfaro**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -323,6 +323,8 @@ dependencies {
     implementation "com.icthh.xm.commons:xm-commons-logging-configurable:${xm_commons_version}"
     implementation "com.icthh.xm.commons:xm-commons-logging-trace:${xm_commons_version}"
     implementation "com.icthh.xm.commons:xm-commons-topic:${xm_commons_version}"
+    implementation "com.icthh.xm.commons:xm-commons-messaging:${xm_commons_version}"
+    implementation "com.icthh.xm.commons:xm-commons-idp:${xm_commons_version}"
```
Comment (sergeysenja1992) [inline build.gradle:327]: dependency duplicate

File: src/main/java/com/icthh/xm/uaa/security/oauth2/idp/config/IdpConfigRepository.java [hunk 2]
```diff
@@ -4,8 +4,8 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
-import com.icthh.xm.commons.domain.idp.IdpConfigUtils;
-import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig;
+import com.icthh.xm.commons.domainevent.idp.IdpConfigUtils;
+import com.icthh.xm.commons.domainevent.idp.model.IdpPublicConfig;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/security/oauth2/idp/config/IdpConfigRepository.java:8]: O_o strange package renaming 

@amedvedchuk @EvgeniyMurenko

File: src/main/java/com/icthh/xm/uaa/service/mail/MailService.java [hunk 3]
```diff
@@ -71,6 +75,11 @@ public class MailService {
     private final TenantContextHolder tenantContextHolder;
     private final TenantConfigService tenantConfigService;
     private final LocalizationMessageService localizationMessageService;
+    private final CommunicationService communicationService;
+    private final ObjectMapper objectMapper;
+
+    @Value("${application.communication.enabled}")
+    private Boolean sendByCommunication;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/service/mail/MailService.java:82]: use variable from ApplicationProperties to make this class more test friendly

File: src/main/resources/config/application.yml [hunk 4]
```diff
@@ -211,6 +211,8 @@ application:
         enablePasswordHashCaching: true
         enablePasswordHashCacheSize: 1000
     last-login-date-enabled: false
+    communication:
+        enabled: true
```
Comment (sergeysenja1992) [inline src/main/resources/config/application.yml:215]: default false

File: src/main/java/com/icthh/xm/uaa/service/mail/MailService.java [hunk 5]
```diff
@@ -44,6 +29,25 @@
 import org.springframework.stereotype.Service;
 import org.springframework.ui.freemarker.FreeMarkerTemplateUtils;
 
+import javax.annotation.Resource;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/service/mail/MailService.java:32]: just sort import without separate section for java imports

**Review by pkev777 — APPROVED** [review]

**Review by Yurwar — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

File: src/main/java/com/icthh/xm/uaa/service/mail/MailService.java (hunk 5 above)
Comment (monassfaro) [reply src/main/java/com/icthh/xm/uaa/service/mail/MailService.java]: Fixed imports

**Pull request merged**